common4j/src/main/com/microsoft/identity/common/java/interfaces/IPerSeparatorMultiTypeNameValueStore.java

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.java.interfaces;
+
+import com.microsoft.identity.common.java.util.ported.Predicate;
+
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+
+import edu.umd.cs.findbugs.annotations.Nullable;
+import lombok.NonNull;
+
+/**
+ * This is a storage that allows to store name-value pairs associated to a particular "Separator".
```

Comment:
nit: I think the term "Storage" (or something along that line) might be easier to understand (yeah... it might be confusing as we're already using the term "store")

This structure consists of multiple key-value storages.

When a dev is using this interface, they're trying to read/write a value from a certain storage.

Maybe we can call this MultiStorageNameValueStore?

Reply:
The idea is that at the interface level the dev shouldn't care if there about the internal implementation details. It is just data separated by a separator and that separator can be anything. If you look at `IAccountDataStorage` - that is the same code basically except separator there is replaced with `IBrokerAccount`: https://github.com/AzureAD/ad-accounts-for-android/blob/dev/broker4j/src/main/com/microsoft/identity/broker4j/broker/platform/components/IAccountDataStorage.java#L301-L321

Since that one is called `IAccountDataStorage`, I followed similar naming mechanism here i.e. ISeparatorNameValueStore or something along those lines.

I can definitely rename it from Store to Storage.